zstd.cc

```diff
@@ -23,116 +23,106 @@ static const sstring COMPRESSOR_NAME = compressor::namespace_prefix + "ZstdCompr
 static const size_t DCTX_SIZE = ZSTD_estimateDCtxSize();
 
 class zstd_processor : public compressor {
```

Comment:
These changes in `zstd_processor` look like they belong to a separate commit, not the one titled "Add cache to named/id lookup of compressors".

They seem to be much more than refactors, the logic appears to change significantly.

Reply:
These changes need an elaborate commit message, because TBH I don't understand what's happening / the motivation.